src/AutoRest.CSharp.V3/Output/Builders/ClientBuilder.cs

```diff
@@ -270,28 +285,32 @@ private static Method BuildNextPageMethod(Method method)
                 method.Diagnostics);
         }
 
-        private static Paging GetClientMethodPaging(Method method, Method nextPageMethod, IDictionary<object, object> pageable, ObjectSchema? schema)
+        private Paging GetClientMethodPaging(Method method, Method nextPageMethod, IDictionary<object, object> pageable, ObjectType type)
         {
-            var nextLinkName = pageable.GetValue<string>("nextLinkName");
-            var itemName = pageable.GetValue<string>("itemName");
-            //TODO: Hack to figure out the property name on the model
-            var itemProperty = schema?.Properties?.FirstOrDefault(p => p.SerializedName == itemName);
-            itemName = itemProperty?.CSharpName() ?? itemName ?? "Value";
-            var nextLinkProperty = schema?.Properties?.FirstOrDefault(p => p.SerializedName == nextLinkName);
-            nextLinkName = nextLinkProperty?.CSharpName() ?? nextLinkName;
-            // If itemName resolves to Value, we can't use itemProperty. So, get the correct property.
-            var itemTypeProperty = schema?.Properties?.FirstOrDefault(p => p.CSharpName() == itemName);
-            var itemTypeValueSchema = (itemTypeProperty?.Schema as ArraySchema)?.ElementType;
-            var itemType = BuilderHelpers.CreateType(itemTypeValueSchema ?? new Schema(), false);
-            var name = $"{method.Name}Pageable";
-            return new Paging(method, nextPageMethod, name, nextLinkName, itemName, itemType);
+            var nextLinkName = pageable.GetValue<string>("nextLinkName") ?? "nextLink";
```

Comment:
This is a bug. We cannot default to `nextLink` when no name is provided. There are tests specifically for that.